R/teal_data-class.R

```diff
@@ -0,0 +1,103 @@
+setOldClass("JoinKeys")
+
+#' Reproducible data class.
+#'
+#' Reproducible data class basing on [`teal.code::qenv-class`].
+#' Don't interact with slots directly, it is suggested to use methods instead
+#' (see `methods(class = "teal_data")`).
+#'
+#' @name teal_data-class
+#' @rdname teal_data-class
+#'
+#' @slot code (`character`) representing code necessary to reproduce the environment.
+#'  To extract the `code` please use [get_code()].
+#' @slot env (`environment`) environment which content was generated by the evaluation
+#'  of the `code` slot. To extract variables from the environment please use [get_var()] or [`[[`].
+#' @slot id (`integer`) random identifier of the code element to make sure uniqueness
+#'  when joining.
+#' @slot warnings (`character`) the warnings output when evaluating the code.
+#' To extract the `warnings` use [get_warnings()].
+#' @slot messages (`character`) the messages output when evaluating the code
```

Comment:
can you inherit roxygen params description from parent class to avoid repetition?

Reply:
There is no `@inheritSlots` roxygen tag unfortunatelly

Reply:
Have you tried `@inheritSection slots` (or rather `Slots`)?

Reply:
Doesn't seem to work properly. Look, added slots doesn't merge with inherited section.
 
<img width="1503" alt="Skärmavbild 2023-10-20 kl  10 39 21" src="https://github.com/insightsengineering/teal.data/assets/6959016/5a1cc7bc-2df2-4a55-98cd-368f20767e84">

Reply:
Indeed. It looks like it's a missing feature in roxygen. Let's leave it then. Thanks for checking